### PR TITLE
fix applying deprecated label

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
the application of deprecated label was not working correctly,
because it doesn't skip old templates with the same name as newly
deployed templates.
Added a condition, which check if template has newer version than is
the version in ssp operator. If yes, template is not set as deprecated
**Which issue(s) this PR fixes**: 
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=2042799

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,6 +31,7 @@ require (
 	kubevirt.io/controller-lifecycle-operator-sdk v0.2.1
 	kubevirt.io/qe-tools v0.1.7
 	sigs.k8s.io/controller-runtime v0.9.3
+	golang.org/x/mod v0.5.0
```
Comment (akrejcir) [inline go.mod:34]: We already import semver package: `github.com/blang/semver/v4`.
But this is ok, you don't need to change this PR.

**Review by akrejcir — COMMENTED** [review]
/lgtm

File: go.mod (hunk 1 above)
Comment (ksimon1) [reply go.mod:34]: I will update it, because its not worth to introduce new dependency

**Pull request merged**